include/behavior_tree_core/tree_node.h

```diff
@@ -41,9 +42,34 @@ enum NodeType
     ACTION_NODE,
     CONDITION_NODE,
     CONTROL_NODE,
-    DECORATOR_NODE
+    DECORATOR_NODE,
+    SUBTREE_NODE,
+    UNDEFINED
 };
 
+inline const char* toStr(const BT::NodeType& type)
+{
+    switch (type)
+    {
+        case NodeType::ACTION_NODE:
+            return "Action";
```

Comment:
For actions and condition I would suggest to print the name instead

Reply:
Look carefully. This print NodeType, not TreeNode.

BodeType is an enum.

Reply:
OK